src/document.cpp

```diff
@@ -618,6 +632,49 @@ rvoe<NoReturnValue> PdfDocument::create_catalog() {
   /Pages {} 0 R
 )",
                    pages_object);
+
+    if(!page_labels.empty()) {
+        buf += R"(  /PageLabels<<
+    /Nums [)";
+        for (const auto &page_label : page_labels) {
+            buf += std::format("{} <<", page_label.page_id);
```

Comment:
This creates a temporary string, fills that and then copies the content to the final location. It should use `std::format_to(app`  instead.